src/odemis/odemisd/start.py

```diff
@@ -722,8 +755,43 @@ def main(args):
             else:
                 # Return when the GUI is done
                 logging.info("Starting the GUI...")
-                subprocess.check_call(["odemis-gui", "--log-level", odemis_config["LOGLEVEL"]])
 
+                gui_start_time = time.time()
+                try:
+                    subprocess.check_call(["odemis-gui", "--log-level", odemis_config["LOGLEVEL"]])
+                except Exception as error:
+                    if time.time() < gui_start_time + 3:
+                        error_code_gui = None
+                        # We don't want to fail because the error output changed.
+                        try:
+                            error_code_gui = error.args[0]  # The first argument is assumed to be the error code
+                        except:
+                            pass
+                        if not error_code_gui or not isinstance(error_code_gui, int) or error_code_gui == 0:
+                            error_code_gui = 126  # Use 126 as default error when no correct error code is found.
+
+                        logging.exception(f"Starting of the GUI failed.")
+                        raise GuiError(f"Starting of the GUI caused the following error: {error}")
+                    else: # The GUI closed more than 3 seconds after starting. A normal startup is assumed.
+                        pass
+
+    except GuiError:
+        try:
+            odemis_gui_state = "ON" if find_window(GUI_WM_CLASS) else "OFF"  # find_window may fail
+        except Exception:
+            logging.error(f"The call 'find_window({GUI_WM_CLASS})' failed.")
+            odemis_gui_state = "invalid"
```

Comment:
if for some reason find_window is broken, or even caused this error weird errors will be returned. I think in no case we want the function which shows fails to be failing. I think we want to prevent causing that mess. @pieleric  Do you agree or do you still prefer to remove the try/except?

Reply:
I wasn't actually talking about the exception block, I was talking about the main try block. I don't think it's needed to look for the window. If the GUI failed to start, you're not going to find a window, ever. So no need to search for it. You can drop the whole try/except block.